atc/resource/resource_instance_fetch_source.go

```diff
@@ -172,25 +172,25 @@ func (s *resourceInstanceFetchSource) Create(ctx context.Context) (VersionedSour
 	)
 	if err != nil {
 		sLog.Error("failed-to-fetch-resource", err)
-		return nil, err
+		return versionedSource, err
```

Comment:
Is there a case here that both versionedSource and err are not nil?

Same for the rest of 3 changes of returning versionedSource instead of nil, is there a specific reason for that?